src/component/mxgraph/renderer/StyleComputer.ts

```diff
@@ -158,7 +158,7 @@ export default class StyleComputer {
   }
 
   computeMessageFlowIconStyle(edge: Edge): string {
-    return `shape=${BpmnStyleIdentifier.MESSAGE_FLOW_ICON};${BpmnStyleIdentifier.IS_INITIATING}=${edge.messageVisibleKind}`;
+    return `shape=${BpmnStyleIdentifier.MESSAGE_FLOW_ICON};${BpmnStyleIdentifier.IS_INITIATING}=${edge.messageVisibleKind === MessageVisibleKind.INITIATING}`;
```

Comment:
**issue (non-blocking):**  
In the BPMN specification, we can have 3 states for the `messageVisibleKind`: unspecified, `initiating` and `non-initiating`.

So we shouldn't add `BpmnStyleIdentifier.IS_INITIATING` to the style if `messageVisibleKind` is unspecified.

![image](https://user-images.githubusercontent.com/4921914/221534688-f814e9c4-8fd9-4c67-876c-eb058a77a0fb.png)

Reply:
`computeMessageFlowIconStyle` is only called when we detect that a message flow icon must be drawn, so the check is not done here but in `insertMessageFlowIconIfNeeded`.

https://github.com/process-analytics/bpmn-visualization-js/blob/94ce4bea816d27c8a4c6cca84912a15d649be916/src/component/mxgraph/BpmnRenderer.ts#L111-L113

With the old and new implementation, once the `bpmn.isInitiating` style is present, an icon is displayed (filled or not). Here, we only change how the information is stored, not when it is stored.

All tests including e2e visual an integration tests pass. They may miss some use cases related to the BPMN specification.
In that case, they must be added (both integration and visual tests) but only because, at present, we may not support the unspecified `messageVisibleKind`.

Reply:
As far as I understand, we cover everything in the tests and in the implementation.
A `MessageFlow` always have the `messageVisibleKind` property set. `NONE` is set when it was undefined in the BPMN diagram.

https://github.com/process-analytics/bpmn-visualization-js/blob/94ce4bea816d27c8a4c6cca84912a15d649be916/src/component/parser/json/converter/DiagramConverter.ts#L152

Reply:
I though when we compute the [classes](https://github.com/process-analytics/bpmn-visualization-js/pull/2550/files#diff-f99bcca3056e94c558bcb468f8a4642b5cc3c1918ef58d776ff3964a1ddededeR72) that we always added `messageVisibleKind` :smile:   
It's ok for me like this :+1: